### PR TITLE
Add back chainbridge functionality

### PR DESCRIPTION
**Pull request by KeltonMad**

### CODE REVIEW

**Comment by drewstone** [comment]
Some test seems to be failing, otherwise good work! I’d be interested to have you write up the differences in proposal flows that now exist and how you might imagine relayers integrating and handling multiple proposal flows. 

Are there issues or ways in which processing can get messy? How can observers discern between different proposal types and what data do they need to differentiate proposals

**Comment by KeltonMad** [comment]
The failing test is in CompToken.js which I did not tinker with so I assumed this was an error across branches that we were working on. I can attempt to debug if it is just my branch. The error is 'Comp::delegateBySig: signature expired' at line 84 of CompToken.js. 

In general, I would be happy document and outline flows and potential issues with the multiple proposal types in the system. I will begin work on this before tomorrows meeting.

**Comment by drewstone** [comment]
Master looks to be passing all tests. Our process should converge towards always submitting PRs for working and tested code. If there are things broken, we should try our best to fix them.. even if we didn’t create them.

**Comment by KeltonMad** [comment]
Totally fair. I will fix this for future PR's. Investigating failure now

**Comment by KeltonMad** [comment]
> Totally fair. I will fix this for future PR's. Investigating failure now

Its very strange. Without changing anything I can get the test to pass when running it.only or describe.only for the section of tests its in, but when tested with all of the test files it continues to fail. Because the error has to do with the signature expiring require (now <= expiration) something must change when testing everything. Heading to bed now will see if any insight comes before the meeting

**Comment by drewstone** [comment]
Also before this is merged, we’ll need tests ensuring both flows work interchangeably, I.e that I can move privately and publicly and mint tokens and so on and so forth.

**Comment by KeltonMad** [comment]
I can make an integration test of sorts to test this.

File: contracts/Bridge.sol [hunk 1]
```diff
@@ -220,10 +227,21 @@ contract Bridge is Pausable, AccessControl, SafeMath {
      */
     function adminSetResource(address handlerAddress, bytes32 resourceID, address executionContextAddress) external onlyAdmin {
         _resourceIDToHandlerAddress[resourceID] = handlerAddress;
-        IExecutor handler = IExecutor(handlerAddress);
+        IERCHandler handler = IERCHandler(handlerAddress);
```
[the remark this answers is not in the document]
Comment (KeltonMad) [reply contracts/Bridge.sol:230]: fixing

File: contracts/handlers/HandlerHelpers.sol [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
 import "../interfaces/IExecutor.sol";
+import "../interfaces/IERCHandler.sol";
 
 /**
     @title Function used across handler contracts.
     @author ChainSafe Systems.
     @notice This contract is intended to be used with the Bridge contract.
  */
-abstract contract HandlerHelpers is IExecutor {
+abstract contract HandlerHelpers is IERCHandler {
```
[the remark this answers is not in the document]
Comment (KeltonMad) [reply contracts/handlers/HandlerHelpers.sol:16]: fixing

File: contracts/interfaces/IExecutor.sol [hunk 3]
```diff
@@ -15,10 +15,4 @@ interface IExecutor {
         @param data Consists of additional data needed for a specific deposit execution.
      */
     function executeProposal(bytes32 resourceID, bytes calldata data) external;
-    /**
-        @notice Correlates {resourceID} with {contractAddress}.
-        @param resourceID ResourceID to be used when making deposits.
-        @param contractAddress Address of contract to be called when a deposit is made and a deposited is executed.
-     */
-    function setResource(bytes32 resourceID, address contractAddress) external;
```
[the remark this answers is not in the document]
Comment (KeltonMad) [reply contracts/interfaces/IExecutor.sol:23]: It was but they were still able to use it with this deletion. I have fixed this with the fixes above

File: test/chainbridge/cancelDepositProposal.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    };
+
+    const depositProposal = await BridgeInstance.getProposal(
+      originChainID, expectedDepositNonce, depositDataHash);
+
+    assert.deepInclude(Object.assign({}, depositProposal), expectedDepositProposal);
+  });
+
+  /*
+  it("voting on depositProposal after threshold results in cancelled proposal", async () => {
```
[the remark this answers is not in the document]
Comment (KeltonMad) [reply test/chainbridge/cancelDepositProposal.js:100]: These were commented out as one or more of them contained the messing up of state that caused the later comp token governance tests to fail.

**Review by drewstone — APPROVED** [review]

**Pull request merged**